iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -118,6 +115,13 @@
     if (splitSize > 0) {
       scan = scan.option(TableProperties.SPLIT_SIZE, String.valueOf(splitSize));
     }
+    // In case of LLAP-based execution we ask Iceberg not to combine multiple fileScanTasks into one split.
+    // This is so that cache affinity can work, and each file(split) is executed/cached on always the same LLAP daemon.
+    MapWork mapWork = LlapHiveUtils.findMapWork((JobConf) conf);
+    if (mapWork != null && mapWork.getCacheAffinity()) {
+      Long openFileCost = splitSize > 0 ? splitSize : TableProperties.SPLIT_SIZE_DEFAULT;
+      scan = scan.option(TableProperties.SPLIT_OPEN_FILE_COST, String.valueOf(openFileCost));
```

Comment:
Why will setting the split size as an open file cost will cause scan to use individual files as splits?

Reply:
I would also need to understand that better, so a comment would probably be helpful here